### PR TITLE
Storage: Fix Bucket logging configuration using library

### LINKED ISSUE #2911: Bucket logging configuration using library
I was wondering whether it is possible to set the bucket logging in the target bucket with the access logs are then stored in the bucket log as it is explained here https://cloud.google.com/storage/docs/access-logs but this is only possible using gsutil. There is the option for XML API here https://cloud.google.com/storage/docs/xml-api/put-bucket-logging but not sure if there is one in this google-cloud-java storage library. If it is possible how do I do it?

### PR DESCRIPTION
**Pull request by pmakani**
Fixes #2911

### CODE REVIEW

**Comment by pmakani** [comment]
@frankyn PTAL.

File: google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 1]
```diff
@@ -2714,4 +2714,38 @@ public void testBucketLocationType() throws ExecutionException, InterruptedExcep
...
+    String logsBucket = RemoteStorageHelper.generateBucketName();
+    String loggingBucket = RemoteStorageHelper.generateBucketName();
+    try {
+      assertNotNull(
+          storage.create(
+              BucketInfo.newBuilder(logsBucket)
+                  .setLocation("us")
+                  .setRetentionPeriod(RETENTION_PERIOD)
+                  .build()));
+      Acl acl = Acl.of(User.ofAllAuthenticatedUsers(), Role.WRITER);
```
Comment (frankyn) [inline google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Use IAM policy instead.

File: google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java [hunk 2]
```diff
@@ -2714,4 +2714,38 @@ public void testBucketLocationType() throws ExecutionException, InterruptedExcep
...
+  @Test
+  public void testBucketLogging() throws ExecutionException, InterruptedException {
+    String logsBucket = RemoteStorageHelper.generateBucketName();
+    String loggingBucket = RemoteStorageHelper.generateBucketName();
+    try {
+      assertNotNull(
+          storage.create(
+              BucketInfo.newBuilder(logsBucket)
+                  .setLocation("us")
+                  .setRetentionPeriod(RETENTION_PERIOD)
```
[the remark this answers is not in the document]
Comment (pmakani) [reply google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Done

File: google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java (hunk 1 above)
Comment (pmakani) [reply google-cloud-clients/google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java]: Done.

**Pull request merged**